### PR TITLE
Now with docker-compose and clean Dockerfile

### PR DESCRIPTION
**Pull request by hammady**
Please check commit messages. Now you can run through docker compose and swarm. Check docker/readme.

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+log/*.log
+public/assets
+tmp
+.git*
```
Comment (mmahalwy) [inline .dockerignore]: nit: New line at the end?

File: Dockerfile [hunk 2]
```diff
@@ -1,61 +1,38 @@
...
+RUN mkdir /var/www && \
+    chown -R www-data /app /var/www /usr/local/bundle
+
+USER www-data
+
+# install a matching bundler to Gemfile.lock
+RUN gem install bundler -v 1.15.3
+
+# install all gems
+ARG env=development
```
Comment (mmahalwy) [inline Dockerfile]: how come development?

File: docker-compose.production.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+version: '3'
+services:
+  api:
+    build:
+      context: .
+      args:
+        - env=production
+        - bundle_opts=--without development test --deployment --clean
+    environment:
+      - SENTRY_DSN=https://www.google.com
```
Comment (mmahalwy) [inline docker-compose.production.yml]: Hmmm..

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+version: '3'
```
Comment (mmahalwy) [inline docker-compose.yml]: What's the difference between this and the `docker-compose.production.yml` file?

File: Dockerfile (hunk 2 above)
Comment (hammady) [reply Dockerfile]: this is the default value if not set

File: docker-compose.production.yml (hunk 3 above)
Comment (hammady) [reply docker-compose.production.yml]: lets remove that SENTRY for now

File: docker-compose.yml (hunk 4 above)
Comment (hammady) [reply docker-compose.yml]: when invoked with no -f files, compose will merge this one and .override.yml. by convention this one is the base file to be merged with others (.production.yml in the case of production)

Comment (hammady) [reply docker-compose.yml]: and .override.yml in the case case of dev, check [this](https://docs.docker.com/compose/extends/) for more details.

File: .dockerignore (hunk 1 above)
Comment (hammady) [reply .dockerignore]: sorry my fault

File: docker-compose.production.yml (hunk 3 above)
Comment (hammady) [reply docker-compose.production.yml]: done

**Pull request merged**